plugins/zfs/zfs.go

```diff
@@ -32,7 +37,7 @@ func (z *Zfs) Description() string {
 	return "Read metrics of ZFS from arcstats, zfetchstats and vdev_cache_stats"
 }
 
-func getTags(kstatPath string) map[string]string {
```

Comment:
OK, I see what you mean, one thing you could do is make `tags` a private variable on the `Zfs` struct. Then you can set `z.tags["pool"] = ...` within this function, and don't return a map in that case.

You can either pass in the `Zfs` struct or make this function have a receiver `(z *Zfs)`